freqtrade/exchange/exchange.py

```diff
@@ -2437,24 +2599,27 @@ async def _async_get_trade_history_time(self, pair: str, until: int,
     async def _async_get_trade_history(self, pair: str,
                                        since: Optional[int] = None,
                                        until: Optional[int] = None,
-                                       from_id: Optional[str] = None) -> Tuple[str, List[List]]:
+                                       from_id: Optional[str] = None,
+                                       stop_on_from_id: Optional[bool] = True,
+                                       ) -> Tuple[str, List[List]]:
         """
         Async wrapper handling downloading trades using either time or id based methods.
         """
 
         logger.debug(f"_async_get_trade_history(), pair: {pair}, "
                      f"since: {since}, until: {until}, from_id: {from_id}")
 
-        if until is None:
-            until = ccxt.Exchange.milliseconds()
+        if self._trades_pagination == 'time':
+            if until is None:
+                until = ccxt.Exchange.milliseconds()
```

Comment:
Why make this conditional?

if until is not set, you'll still want to stop at the time of starting the download - otherwise you risk ending up in an endless loop on active pairs - as there's always new trades to fetch when you do a call.


That'll also simplify further handling - as "until" remains non-optional in `_async_get_trade_history_id()` - and should allow to remove "optionality" handling for that parameter there.